workers/unity/Assets/Playground/Scripts/MonoBehaviours/SpawnCubeCommandReceiver.cs

```diff
@@ -0,0 +1,112 @@
+using Improbable.Gdk.Core.Commands;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using Improbable.Worker.Core;
+using UnityEngine;
+using Transform = Generated.Improbable.Transform.Transform;
+
+#region Diagnostic control
+
+// Disable the "variable is never assigned" for injected fields.
+#pragma warning disable 649
+
+// ReSharper disable PossibleInvalidOperationException
+
+#endregion
+
+namespace Playground.MonoBehaviours
+{
+    public class SpawnCubeCommandReceiver : MonoBehaviour
+    {
+        [Require] private Transform.Requirables.Reader transformReader;
+        [Require] private CubeSpawner.Requirables.CommandRequestHandler commandRequestHandler;
+        [Require] private CubeSpawner.Requirables.Writer writer;
+        [Require] private WorldCommands.Requirables.WorldCommandRequestSender worldCommandRequestSender;
+        [Require] private WorldCommands.Requirables.WorldCommandResponseHandler worldCommandResponseHandler;
+
+        private ILogDispatcher logDispatcher;
+
+        private const string EntityForIdFailedWithMessage = "Create entity (for id {0}) failed with message: \"{1}\"";
+        private const string FailedToReserveEntityIdMessage = "Failed to reserve entity id: {0}";
+
+        private readonly HashSet<long> sentRequestIds = new HashSet<long>();
+
+        public void OnEnable()
+        {
+            commandRequestHandler.OnSpawnCubeRequest += OnSpawnCubeRequest;
+
+            worldCommandResponseHandler.OnReserveEntityIdsResponse += OnEntityIdsReserved;
+            worldCommandResponseHandler.OnCreateEntityResponse += OnEntityCreated;
+
+            logDispatcher = GetComponent<SpatialOSComponent>().LogDispatcher;
+        }
+
+        private void OnSpawnCubeRequest(CubeSpawner.SpawnCube.RequestResponder requestResponder)
+        {
+            requestResponder.SendResponse(new Empty());
+
+            sentRequestIds.Add(worldCommandRequestSender.ReserveEntityIds(1));
+        }
+
+        private void OnEntityIdsReserved(WorldCommands.ReserveEntityIds.ReceivedResponse response)
+        {
+            if (!sentRequestIds.Remove(response.RequestId))
+            {
+                // This response was not for a command from this behaviour.
+                return;
+            }
+
+            var responseOp = response.Op;
+
+            if (responseOp.StatusCode != StatusCode.Success)
+            {
+                logDispatcher.HandleLog(LogType.Error,
+                    new LogEvent(string.Format(FailedToReserveEntityIdMessage, responseOp.Message)));
+
+                return;
+            }
+
+            var location = transformReader.Data.Location;
+            var cubeEntityTemplate =
+                CubeTemplate.CreateCubeEntityTemplate(new Coordinates(location.X, location.Y + 2, location.Z));
+            var expectedEntityId = responseOp.FirstEntityId.Value;
+
+            sentRequestIds.Add(worldCommandRequestSender.CreateEntity(cubeEntityTemplate, expectedEntityId));
```

Comment:
I don't think request IDs are guaranteed to be unique across different worker types

Reply:
But do we guarantee that or is it an implementation detail?

Reply:
I am confident they are unique but I am aware it's not very obvious that they are, but I'll eliminate the problem here by using the context instead